src/Common/ImageTags.tsx

```diff
@@ -0,0 +1,594 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import { ReactComponent as Add } from '../Assets/Icon/ic-add.svg'
+import { ReactComponent as Close } from '../Assets/Icon/ic-cross.svg'
+import { ReactComponent as QuestionFilled } from '../Assets/Icon/ic-help.svg'
+import { ReactComponent as Question } from '../Assets/Icon/ic-help-outline.svg'
+
+import { ReactComponent as EditIcon } from '../Assets/Icon/ic-pencil.svg'
+import { ReactComponent as Redo } from '../Assets/Icon/ic-arrow-counter-clockwise.svg'
+import { ReactComponent as Minus } from '../Assets/Icon/ic-minus.svg'
+import { toast } from 'react-toastify'
+import { ReactComponent as Info } from '../Assets/Icon/ic-info-filled.svg'
+import { ReactComponent as Error } from '../Assets/Icon/ic-warning.svg'
+import { ReactComponent as Warning } from '../Assets/Icon/ic-error-exclamation.svg'
+import { ReactComponent as Enter } from '../Assets/Icon/ic-enter.svg'
+import { ImageButtonType, ImageTaggingContainerType, ReleaseTag, TippyTheme } from './Types'
+import { showError, stopPropagation } from './Helper'
+import { TippyCustomized } from './TippyCustomized'
+import { setImageTags, getUserRole } from './Common.service'
+import Tippy from '@tippyjs/react'
+import { ServerErrors } from './ServerError'
+
+export const ImageTagsContainer = ({
+    ciPipelineId,
+    artifactId,
+    imageComment,
+    imageReleaseTags,
+    updateCurrentAppMaterial,
+    appReleaseTagNames,
+    setAppReleaseTagNames,
+    tagsEditable,
+    setTagsEditable,
+    toggleCardMode,
+    hideHardDelete,
+    forceReInit,
+}: ImageTaggingContainerType) => {
+    const [initialTags, setInitialTags] = useState<ReleaseTag[]>(imageReleaseTags ? imageReleaseTags : [])
+    const [initialDescription, setInitialDescription] = useState(imageComment ? imageComment.comment : '')
+    const [existingTags, setExistingTags] = useState(appReleaseTagNames ? appReleaseTagNames : [])
+    const [newDescription, setNewDescription] = useState(imageComment ? imageComment.comment : '')
+    const [isEditing, setIsEditing] = useState(false)
+    const [showTagsWarning, setShowTagsWarning] = useState(false)
+    const [displayedTags, setDisplayedTags] = useState<ReleaseTag[]>(imageReleaseTags ? imageReleaseTags : [])
+    const [tagErrorMessage, setTagErrorMessage] = useState('')
+    const [createTags, setCreateTags] = useState<ReleaseTag[]>([])
+    const [softDeleteTags, setSoftDeleteTags] = useState<ReleaseTag[]>([])
+    const [hardDeleteTags, setHardDeleteTags] = useState<ReleaseTag[]>([])
+    const [isSuperAdmin, setSuperAdmin] = useState<boolean>(false)
+    const [descriptionValidationMessage, setDescriptionValidationMessage] = useState<string>('')
+    const [textInput, setTextInput] = useState<string>('')
+    const [saveData, setSaveData] = useState<boolean>(false)
```

Comment:
We can remove this variable. can be handled without this

Reply:
done